### PR TITLE
feat: Adding commitlint and husky

### PR DESCRIPTION
**Pull request by filipdanisko**
Adding commitlint, spliting #1661 to separate MR as we are not sure yet about the changelog stuff
<br/><br/><br/><url>LiveURL: https://orbit-components-update-commit-lint.surge.sh</url>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -89,7 +95,10 @@
   },
   "dependencies": {
     "@adeira/js": "^1.2.2",
-    "@kiwicom/orbit-design-tokens": "~0.9.0"
+    "@commitlint/cli": "^8.3.5",
```
Comment (vepor) [inline package.json]: everything should be dev deps 🙏 @filipdanisko

File: package.json [hunk 2]
```diff
@@ -78,6 +78,12 @@
       }
     }
   },
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
+      "pre-push": "yarn prettier:fix"
```
Comment (vepor) [inline package.json]: do we really want prettier:fix runned on pre-push? I would have it also for commit

**Comment by vepor** [comment]
Also, can you remove prettier:check from CI?

File: package.json (hunk 2 above)
Comment (filipdanisko) [reply package.json]: Well `commit-msg` is validated after `pre-commit` making it little painfull to wait for prettier to run before commit is validated, it still takes couple of seconds which is totally ok for push, not so sure about commit though

**Pull request merged**